pointcloud/src/main/scala/geotrellis/pointcloud/raster/ept/DEMRasterSource.scala

```diff
@@ -102,16 +102,124 @@ case class DEMRasterSource(
         pipeline.execute
 
         val pointViews = pipeline.getPointViews().asScala.toList
-        val viewSizes = pointViews.map(_.length)
-        val (nCoords, starts) = viewSizes.foldLeft(0 -> List.empty[Int]) { case ((acc, cum), v) => (v + acc, cum :+ acc) }
-        val coords = Array.ofDim[Coordinate](nCoords)
-        viewSizes.zip(starts).zip(pointViews).foreach { case ((n, z), pv) =>
-          val pc = pv.getPointCloud(DimType.X, DimType.Y, DimType.Z)
-          cfor(0)(_ < n, _ + 1) { i => coords(z + i) = pc.getCoordinate(i) }
+        assert(pointViews.length == 1, "Triangulation pipeline should have single resulting point view")
+
+        val pc = pointViews(0).getPointCloud(DimType.X, DimType.Y, DimType.Z)
+        val tris = pointViews(0).getTriangularMesh()
+
+        val w = gridExtent.cellwidth
+        val h = gridExtent.cellheight
+        val cols = bounds.width
+        val rows = bounds.height
+        val tile = DoubleArrayTile.empty(cols.toInt, rows.toInt)
+
+        def rasterizeTriangle(tri: Triangle): Unit = {
+          val (a,b,c) = (tri.a, tri.b, tri.c)
```

Comment:
Btw, [Triangle](https://github.com/PDAL/java/blob/2.1.4/core/src/main/scala/io/pdal/Triangle.scala#L37-L46) is isomorphic to tuple3.

Reply:
And yet,
```scala
val (a,b,c) = tri
```
Failed to compile.  Probably better to use
```scala
val Triangle(a,b,c) = tri
```